hearth/lib/hearth/signers.rb

```diff
@@ -1,11 +1,16 @@
 # frozen_string_literal: true
 
 module Hearth
+  # Namespace for all Signer classes.
   module Signers
     # Base class for all Signer classes.
     class Base
-      def sign(request:, identity:, properties: {})
-        # do nothing
+      def sign(request:, identity:, properties:)
+        # Do nothing by default
+      end
+
+      def reset(request:, properties:)
+        # Do nothing by default
```

Comment:
Do we want the base signer to do nothing here or should it raise a not implemented error - to suggest that you should provide an implementation (even if its empty)?